migrations/versions/0315_document_count.py

```diff
@@ -0,0 +1,26 @@
+"""
+
+Revision ID: 0315_document_count
+Revises: 0314_populate_email_access
+Create Date: 2020-02-12 14:19:18.066425
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+revision = '0315_document_count'
+down_revision = '0314_populate_email_access'
+
+
+def upgrade():
+    op.add_column('notifications', sa.Column('document_count', sa.Integer(), nullable=True))
+    op.create_index(op.f('ix_notifications_document_count'), 'notifications', ['document_count'], unique=False)
```

Comment:
From what I've been reading recently (see https://medium.com/braintree-product-technology/postgresql-at-scale-database-schema-changes-without-downtime-20d3749ed680#04b4 and section on `Create index`), this will hold an `ACCESS EXCLUSIVE` lock and mean no other other queries, including select statements can access the table which this migration happens. https://www.postgresql.org/docs/9.1/explicit-locking.html

Given the size of the notification and notification history tables, this has the potential to block all requests to the database for a long duration. There is a chance that because we are creating an index on a column that will just be nulls if the speed of creating the index will be super speedy and so it wouldn't matter anyway or if it may still cause a long running transaction.

You could look into creating the index concurrently as suggested in the first link.

Reply:
update: removed index as decided it's not necessary yet